src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String COMMA = ",";
+    private static final String RESULT = "result";
+    private int sumOfSupply = 0;
+    private int sumOfBuy = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        readDataFromFile(fromFileName);
+        String report = createReport(sumOfSupply, sumOfBuy);
+        writeDataToFile(toFileName, report);
+        resetCalculations();
```

Comment:
The `getStatistic` method is doing too much by reading data, creating a report, writing to a file, and resetting calculations. Consider dividing this logic into separate methods to adhere to the principle that one method should be responsible for one task.